Gems/AtomTressFX/Assets/Shaders/HairRenderingSrgs.azsli

```diff
@@ -0,0 +1,208 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// This is code decoration for our sample.  
+// Remove this and the EndHLSL at the end of this file 
+// to turn this into valid HLSL
+
+// originated from: TressFXRendering.hlsl
+
+#pragma once
+
+#include <viewsrg.srgi>
+
+//------------------------------------------------------------------------------
+#define g_mVP               ViewSrg::m_viewProjectionMatrix
+#define g_vEye              ViewSrg::m_worldPosition.xyz
+#define g_vViewport         ViewSrg::m_unprojectionConstants    // xy = normaliz scale, zw = offset in 0..1 scale
+#define cInvViewProjMatrix  ViewSrg::m_viewProjectionInverseMatrix
+//------------------------------------------------------------------------------
+
+#define AMD_PI 3.1415926
+#define AMD_e 2.71828183
+#define AMD_TRESSFX_KERNEL_SIZE 5
+#define KBUFFER_SIZE 8
+#define MAX_FRAGMENTS 512
+// Using value 0 as indicator will skip the first hair fragment to allow 
+// for clear screen value (0). 
+// If Atom will allow other value (0xffffffff) this can be avoided.
+#define FRAGMENT_LIST_NULL 0    
+//!=============================================================================
+//!
+//!                             Render Pass only
+//! 
+//!=============================================================================
+// If you change this, you MUST also change TressFXRenderParams in TressFXConstantBuffers.h
+// org: cbuffer TressFXParameters : register(b3, space0)
+struct TressFXRenderParameters
+{
+    // General information
+    float       m_hairFiberRadius;
+
+    // For deep approximated shadow lookup
+    float       m_shadowAlpha;
+    float       m_fiberSpacing;
+
+    // Original TressFX Kajiya lighting model parameters
+    float       m_hairKs2;
+    float       m_hairEx2;
+    float3      m_fPadding0;           
+    float4      m_matKValue;
+
+    int         m_maxShadowFibers;
+
+    // Marschner lighting model parameters 
+    float       m_roughness;
+    float       m_cuticleTilt;
+
+    float       m_fPadding1;
+};
+
+// Separate strand params from pixel render params (so we can index for PPLL)
+// If you change this, you MUST change TressFXStrandParams in TressFXConstantBuffers.h
+// org: cbuffer TressFXStrandParameters : register(b4, space0)
+struct TressFXStrandParameters
+{
+    float4      m_matBaseColor;
+    float4      m_matTipColor;
+
+    float       m_tipPercentage;
+    float       m_strandUVTilingFactor;
+    float       m_fiberRatio;
+    float       m_fiberRadius;
+
+    int         m_numVerticesPerStrand;
+    int         m_enableThinTip;
+    int         m_nodePoolSize;
+    int         m_renderParamsIndex;    // Material index in the hair material array
+
+    // Other params
+    int         m_enableStrandUV;
+    int         m_enableStrandTangent;
```

Comment:
You can't use shader options here since shader options work as global selector and this is a hair-specific choice.  In other words, when several hair objects exist, each can have different options and the render is using deferred mode, hence you do not operate on per-object basis.